packages/smooth_app/lib/pages/scan/ml_kit_scan_page.dart

```diff
@@ -0,0 +1,285 @@
+                  const Spacer(),
+                  SmoothProductCarousel(
+                    showSearchCard: true,
+                    height: carouselHeight,
+                  ),
+                ],
+              ),
+            ),
+          ],
+        );
+      },
+    );
+  }
+
+  Future<void> _startLiveFeed() async {
+    final CameraDescription camera = cameras[_cameraIndex];
+    _controller = CameraController(
+      camera,
+      ResolutionPreset.high,
+      enableAudio: false,
+    );
+    _controller?.initialize().then((_) {
+      if (!mounted) {
+        return;
+      }
+      _controller?.setFocusMode(FocusMode.auto);
+      _controller?.lockCaptureOrientation(DeviceOrientation.portraitUp);
+      _controller?.startImageStream(_processCameraImage);
+      setState(() {});
+    });
+  }
+
+  Future<void> _stopLiveFeed() async {
+    await _controller?.stopImageStream();
+    _controller = null;
+  }
+
+  Future<void> _disposeLiveFeed() async {
+    await _controller?.dispose();
+  }
+
+  //Convert the [CameraImage] to a [InputImage]
+  Future<void> _processCameraImage(CameraImage image) async {
+    final WriteBuffer allBytes = WriteBuffer();
+    for (final Plane plane in image.planes) {
+      allBytes.putUint8List(plane.bytes);
+    }
+    final Uint8List bytes = allBytes.done().buffer.asUint8List();
+
+    final Size imageSize =
+        Size(image.width.toDouble(), image.height.toDouble());
+
+    final CameraDescription camera = cameras[_cameraIndex];
+    final InputImageRotation imageRotation =
+        InputImageRotationMethods.fromRawValue(camera.sensorOrientation) ??
+            InputImageRotation.Rotation_0deg;
+
+    final InputImageFormat inputImageFormat =
+        InputImageFormatMethods.fromRawValue(
+              int.parse(image.format.raw.toString()),
+            ) ??
+            InputImageFormat.NV21;
+
+    final List<InputImagePlaneMetadata> planeData = image.planes.map(
+      (Plane plane) {
+        return InputImagePlaneMetadata(
+          bytesPerRow: plane.bytesPerRow,
+          height: plane.height,
+          width: plane.width,
+        );
+      },
+    ).toList();
+
+    final InputImageData inputImageData = InputImageData(
+      size: imageSize,
+      imageRotation: imageRotation,
+      inputImageFormat: inputImageFormat,
+      planeData: planeData,
+    );
+
+    final InputImage inputImage =
+        InputImage.fromBytes(bytes: bytes, inputImageData: inputImageData);
+
+    _scanImage(inputImage);
+  }
+
+  //Checking for barcodes in the provided InputImage
+  Future<void> _scanImage(InputImage inputImage) async {
+    if (barcodeScanner == null || isBusy) {
+      return;
+    }
+
+    isBusy = true;
+
+    final List<Barcode> barcodes =
+        await barcodeScanner!.processImage(inputImage);
+
+    //ignore: avoid_function_literals_in_foreach_calls
+    barcodes.forEach((Barcode barcode) {
+      _model.onScan(barcode.value.rawValue);
+    });
```

Comment:
I'm sure you can write a `for` loop.

Reply:
It somehow had a problem with both so I just stuck with this
![Screenshot 2022-01-07 170126](https://user-images.githubusercontent.com/39344769/148687410-b65ffa4a-5073-4f30-ac1c-c147e03a6ed5.png)

Reply:
I think they meant `barcodes.forEach(_model.onScanMLKit);`

Reply:
Then we would have to pass Barcode

Reply:
That's what you're doing in this part of the code: you pass `Barcode barcode`.

Reply:
Ohh this was actually a old screenshot now I am passing `barcode.value.rawValue`, with this and for it works